src/AWS.Deploy.Common/Recipes/Validation/OptionSettingItemValidators/FileValidator.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.IO;
+using AWS.Deploy.Common.IO;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// Validates that a recipe or deployment bundle option with a FilePath typehint points to an actual file.
+    /// This can either be an absolute path to the file or relative to the project path for the current recommendation.
```

Comment:
nit: im not sure what you mean by `for the current recommendation` but i propose removing it.

Reply:
Dropped. Early on I was trying to clarify that it's the recommendation that the user has selected from the list, but having worked on this project longer it seems like that's implied most places.